app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsFragment.kt

```diff
@@ -0,0 +1,681 @@
+                viewModel.accept(InfallibleUiAction.SaveVisibleId(visibleId = id))
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        NotificationHelper.clearNotificationsForActiveAccount(requireContext(), accountManager)
+    }
+
+    override fun onReply(position: Int) {
+        val status = adapter.peek(position)?.statusViewData?.status ?: return
+        super.reply(status)
+    }
+
+    override fun onReblog(reblog: Boolean, position: Int) {
+        val statusViewData = adapter.peek(position)?.statusViewData ?: return
+        viewModel.accept(StatusAction.Reblog(reblog, statusViewData))
+    }
+
+    override fun onFavourite(favourite: Boolean, position: Int) {
+        val statusViewData = adapter.peek(position)?.statusViewData ?: return
+        viewModel.accept(StatusAction.Favourite(favourite, statusViewData))
+    }
+
+    override fun onBookmark(bookmark: Boolean, position: Int) {
+        val statusViewData = adapter.peek(position)?.statusViewData ?: return
+        viewModel.accept(StatusAction.Bookmark(bookmark, statusViewData))
+    }
+
+    override fun onVoteInPoll(position: Int, choices: List<Int>) {
+        val statusViewData = adapter.peek(position)?.statusViewData ?: return
+        val poll = statusViewData.status.poll ?: return
+        viewModel.accept(StatusAction.VoteInPoll(poll, choices, statusViewData))
+    }
+
+    override fun onMore(view: View, position: Int) {
+        val status = adapter.peek(position)?.statusViewData?.status ?: return
+        super.more(status, view, position)
+    }
+
+    override fun onViewMedia(position: Int, attachmentIndex: Int, view: View?) {
+        val status = adapter.peek(position)?.statusViewData?.status ?: return
+        super.viewMedia(attachmentIndex, list(status), view)
+    }
+
+    override fun onViewThread(position: Int) {
+        val status = adapter.peek(position)?.statusViewData?.status ?: return
+        super.viewThread(status.actionableId, status.actionableStatus.url)
+    }
+
+    override fun onOpenReblog(position: Int) {
+        val account = adapter.peek(position)?.account!!
+        onViewAccount(account.id)
+    }
+
+    override fun onExpandedChange(expanded: Boolean, position: Int) {
+        val notificationViewData = adapter.snapshot()[position] ?: return
+        notificationViewData.statusViewData = notificationViewData.statusViewData?.copy(
+            isExpanded = expanded
+        )
+        adapter.notifyItemChanged(position)
+    }
+
+    override fun onContentHiddenChange(isShowing: Boolean, position: Int) {
+        val notificationViewData = adapter.snapshot()[position] ?: return
+        notificationViewData.statusViewData = notificationViewData.statusViewData?.copy(
+            isShowingContent = isShowing
+        )
+        adapter.notifyItemChanged(position)
+    }
+
+    override fun onLoadMore(position: Int) {
+        // Empty -- this fragment doesn't show placeholders
```

Comment:
I've seen it do.

(Update after a resume with lots of notifications inbetween. They are added at the top, the last is a placeholder.)

Reply:
Or does / should this change with this PR?

Reply:
From the first sentence of the PR description:

> "Load more" is gone

Further in the description:

> With no need for "Load more" support, the distinction between ".Concrete" and ".Placeholder" types is unnecessary, the ".Placeholder" type has been removed, and ".Concrete" merged in to "Notification".